dynamic_rest/serializers.py

```diff
@@ -301,6 +317,29 @@ def get_name(cls):
         return cls.Meta.name
 
     @classmethod
+    def get_url(self, pk=None):
+        # if associated with a registered viewset, use its URL
+        url = getattr(self, '_url', None)
```

Comment:
This will only be set in the router if the serializer is associated directly with an exposed view. If the serializer has no view (e.g. a relationship that isn't exposed as "top-level" or a serializer that is used conditionally by a polymorphic view), then the canonical path (via serializer's model) will be used.

From the perspective of a serializer, I think its more intuitive that the URL is primarily derived from the routing hierarchy (API level), with a fallback to the underlying model context (ORM level). This is because the routing hierarchy will unambiguously resolve any connected serializer, whereas the model context will not.

This is particularly apparent in the (common) multi-version scenario. Lets say we have a V1 and a V2 representation of users, which means two distinct sets of urls/views/serializers that reference a common model. If we take V2 to be canonical after the release of V2, suddenly V1's serializer will start returning references to V2. If we take V1 to be canonical, we have the same issue in the other direction.